docs/source/blockarray.rst

```diff
@@ -0,0 +1,198 @@
+ * When a :class:`.BlockArray` is passed to a reduction function, the blocks are
+   ravelled (i.e., reshaped to be 1D) and concatenated before the reduction
+   is applied. This behavior may be prevented by passing the `axis`
+   argument, in which case the function is mapped over the blocks.
+ * When one or more :class:`.BlockArray` instances are passed to a mathematical
+   function that is not a reduction, the function is mapped over
+   (corresponding) blocks.
+
+For a list of array creation routines, see
+
+  ::
+
+    >>> scico.numpy.creation_routines  # doctest: +ELLIPSIS
+    ('empty', ...)
+
+For a list of  reduction functions, see
+
+  ::
+
+    >>> scico.numpy.reduction_functions  # doctest: +ELLIPSIS
+    ('sum', ...)
+
+For lists of the remaining wrapped functions, see
+
+  ::
+
+    >>> scico.numpy.mathematical_functions  # doctest: +ELLIPSIS
+    ('sin', ...)
+    >>> scico.numpy.testing_functions  # doctest: +ELLIPSIS
+    ('testing.assert_allclose', ...)
+    >>> import scico.scipy
+    >>> scico.scipy.special.functions  # doctest: +ELLIPSIS
+    ('betainc', ...)
+
+
+Motivating Example
+------------------
+
+Consider a two-dimensional array :math:`\mb{x} \in \mbb{R}^{n \times m}`.
+
+We compute the discrete differences of :math:`\mb{x}` in the horizontal
+and vertical directions, generating two new arrays: :math:`\mb{x}_h \in
+\mbb{R}^{n \times (m-1)}` and :math:`\mb{x}_v \in \mbb{R}^{(n-1)
+\times m}`.
+
+As these arrays are of different shapes, we cannot combine them into a
+single :class:`~numpy.ndarray`. Instead, we might vectorize each array and concatenate
+the resulting vectors, leading to :math:`\mb{\bar{x}} \in
+\mbb{R}^{n(m-1) + m(n-1)}`, which can be stored as a one-dimensional
+:class:`~numpy.ndarray`. Unfortunately, this makes it hard to access the individual
+components :math:`\mb{x}_h` and :math:`\mb{x}_v`.
+
+Instead, we can form a :class:`.BlockArray`: :math:`\mb{x}_B =
+[\mb{x}_h, \mb{x}_v]`
+
+
+  ::
+
+    >>> n = 32
+    >>> m = 16
+    >>> x_h, key = scico.random.randn((n, m-1))
+    >>> x_v, _ = scico.random.randn((n-1, m), key=key)
+
+    # Form the blockarray
+    >>> x_B = snp.blockarray([x_h, x_v])
+
+    # The blockarray shape is a tuple of tuples
+    >>> x_B.shape
+    ((32, 15), (31, 16))
+
+    # Each block component can be easily accessed
+    >>> x_B[0].shape
+    (32, 15)
+    >>> x_B[1].shape
+    (31, 16)
+
+
+Constructing a BlockArray
+-------------------------
+
+The recommended way to construct a :class:`.BlockArray` is by using the
+`snp.blockarray` function.
+
+  ::
+
+     >>> import scico.numpy as snp
+     >>> x0, key = scico.random.randn((32, 32))
+     >>> x1, _ = scico.random.randn((16,), key=key)
+     >>> X = snp.blockarray((x0, x1))
+     >>> X.shape
+     ((32, 32), (16,))
+     >>> X.size
+     (1024, 16)
+     >>> len(X)
+     2
+
+While :func:`.snp.blockarray` will accept either :class:`~numpy.ndarray` or
```

Comment:
@Michael-T-McCann: This sentence (specifically the "backed by" phrasing) seems like a hold-over from the previous `BlockArray` implementation.

Reply:
Updated. I'm not sure about whether a copy is involved in construction, so I removed that text. I would suspect there is not a copy.